driver/config/provider.go

```diff
@@ -397,6 +404,26 @@ func (p *DefaultProvider) fallbackURL(ctx context.Context, path string, host str
 	return &u
 }
 
+// GetDeviceAndUserCodeLifespan returns the device_code and user_code lifespan. Defaults to 15 minutes.
+func (p *DefaultProvider) GetDeviceAndUserCodeLifespan(ctx context.Context) time.Duration {
+	return p.p.DurationF(KeyDeviceAndUserCodeLifespan, time.Minute*15)
+}
+
+// GetDeviceAuthTokenPollingInterval returns device grant token endpoint polling interval. Defaults to 5 seconds.
+func (p *DefaultProvider) GetDeviceAuthTokenPollingInterval(ctx context.Context) time.Duration {
+	return p.p.DurationF(KeyDeviceAuthTokenPollingInterval, time.Second*5)
+}
+
+// GetUserCodeLength returns configured user_code length
+func (c *DefaultProvider) GetUserCodeLength(ctx context.Context) int {
+	return 8
+}
+
+// GetDeviceAuthTokenPollingInterval returns configured user_code allowed symbols
+func (c *DefaultProvider) GetUserCodeSymbols(ctx context.Context) []rune {
+	return []rune(randx.AlphaUpper)
```

Comment:
The RFC recommends: https://datatracker.ietf.org/doc/html/rfc8628#section-6.1

What do you think about offering security profiles here?

- alpha-upper
- alpha-upper-lower
- numeric-alpha-upper-lower
- numeric-alpha-upper
- numeric

or alternatively

```
entropy: high | medium | low
```

where high is case sensitive alpha numeric (8 length) medium is alpha upper (lenght 8) and low is numeric (length 9)

Reply:
I am not sure if we want to mix upper and lower alpha or alpha an numeric, as the spec states it may be hard for the user to differentiate `l` from `I` or `O` from `0`. 

Currently the entropy we get is 26^8, which is more than what is recommended in the RFC. But I guess some users may want to tinker with this config.

I think I prefer the security profiles you proposed to the entropy config. I will push an update for this later.

Reply:
> I am not sure if we want to mix upper and lower alpha or alpha an numeric, as the spec states it may be hard for the user to differentiate l from I or O from 0.

In my view it really depends on the font, but you're right that they officially discourage it. How about:

- high: alpha-upper 9 chars
- medium: alpha-upper 8 chars
- low: numeric 9 chars

also open to other suggestions.

Reply:
https://github.com/ory/x/pull/836

Reply:
Nice, I will change it to:
- high: alpha-num-no-ambiguous 8 chars
- medium: alpha-upper 8 chars
- low: numeric 9 chars